includes/Abilities/Excerpt_Generation/Excerpt_Generation.php

```diff
@@ -0,0 +1,246 @@
+		return array(
+			'type'       => 'object',
+			'properties' => array(
+				'content' => array(
+					'type'              => 'string',
+					'sanitize_callback' => 'sanitize_text_field',
+					'description'       => esc_html__( 'Content to generate excerpt suggestions for.', 'ai' ),
+				),
+				'post_id' => array(
+					'type'              => 'integer',
+					'sanitize_callback' => 'absint',
+					'description'       => esc_html__( 'Content from this post will be used to generate excerpt suggestions. This overrides the content parameter if both are provided.', 'ai' ),
+				),
+			),
+		);
+	}
+
+	/**
+	 * Returns the output schema of the ability.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return array<string, mixed> The output schema of the ability.
+	 */
+	protected function output_schema(): array {
+		return array(
+			'type'       => 'object',
+			'properties' => array(
+				'excerpt' => array(
+					'type'        => 'string',
+					'description' => esc_html__( 'Generated excerpt.', 'ai' ),
+				),
+			),
+		);
+	}
+
+	/**
+	 * Executes the ability with the given input arguments.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param mixed $input The input arguments to the ability.
+	 * @return array{excerpt: string}|\WP_Error The result of the ability execution, or a WP_Error on failure.
+	 */
+	protected function execute_callback( $input ) {
+		// Default arguments.
+		$args = wp_parse_args(
+			$input,
+			array(
+				'content' => null,
+				'post_id' => null,
+			),
+		);
+
+		// If a post ID is provided, ensure the post exists before using its' content.
+		if ( $args['post_id'] ) {
+			$post = get_post( $args['post_id'] );
+
+			if ( ! $post ) {
+				return new WP_Error(
+					'post_not_found',
+					/* translators: %d: Post ID. */
+					sprintf( esc_html__( 'Post with ID %d not found.', 'ai' ), absint( $args['post_id'] ) )
+				);
+			}
+
+			// Get the post context.
+			$context = get_post_context( $args['post_id'] );
+
+			// Default to the passed in content if it exists.
+			if ( $args['content'] ) {
+				$context['content'] = normalize_content( $args['content'] );
+			}
+		} else {
+			$context = array(
+				'content' => normalize_content( $args['content'] ?? '' ),
+			);
+		}
+
+		// If we have no content, return an error.
+		if ( empty( $context['content'] ) ) {
+			return new WP_Error(
+				'content_not_provided',
+				esc_html__( 'Content is required to generate excerpt suggestions.', 'ai' )
+			);
+		}
+
+		// Generate the excerpts.
+		$result = $this->generate_excerpt( $context );
+
+		// If we have an error, return it.
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		// If we have no results, return an error.
+		if ( empty( $result ) ) {
+			return new WP_Error(
+				'no_results',
+				esc_html__( 'No excerpt suggestions were generated.', 'ai' )
+			);
+		}
+
+		// Return the excerpts in the format the Ability expects.
+		return array(
+			'excerpt' => sanitize_textarea_field( trim( $result[0], ' "\'' ) ),
+		);
```

Comment:
If we change the output schema as mentioned above, we'll need to change this as well:
```suggestion
		return sanitize_textarea_field( trim( $result[0], ' "\'' ) );
```

Reply:
This has been updated